pkg/api/webhook_test.go

```diff
@@ -191,4 +225,77 @@ var _ = ginkgo.Describe("Test Sync Canary Job", ginkgo.Ordered, func() {
 		Expect(err).To(BeNil())
 		Expect(result.DeletedAt).To(Not(BeNil()))
 	})
+
+	ginkgo.It("Should create 1 new check from the webhook and deleted one", func() {
+		body := `
```

Comment:
We need to split this into multiple individual calls to test how the webhook behaves after multiple subsequent calls with data that differs from the first. 
3 x firing
1 x resolved

and then verify 2 exist and 1 is deleted